test/unit/model/implementation/TestRunnerMessageServiceTest.php

```diff
@@ -0,0 +1,132 @@
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoProctoring\test\unit\model\implementation;
+
+use oat\oatbox\service\ServiceManager;
+use oat\tao\model\featureFlag\FeatureFlagCheckerInterface;
+use oat\taoProctoring\model\implementation\TestRunnerMessageService;
+use qtism\runtime\tests\AssessmentTestSession;
+use PHPUnit\Framework\TestCase;
+
+final class TestRunnerMessageServiceTest extends TestCase
+{
+    /**
+     * @var TestRunnerMessageService
+     */
+    private $testRunnerMessageService;
+
+    /**
+     * @var ServiceManager
+     */
+    private $serviceManager;
+
+    /**
+     * @var FeatureFlagCheckerInterface
+     */
+    private $featureFlagChecker;
+
+    /**
+     * @var AssessmentTestSession
+     */
+    private $assessmentTestSession;
+
+    protected function setUp(): void
+    {
+        $this->testRunnerMessageService = new TestRunnerMessageService();
+        $this->featureFlagChecker = $this->createMock(FeatureFlagCheckerInterface::class);
+        $this->serviceManager = $this->createMock(ServiceManager::class);
+        $this->serviceManager->method('get')->willReturn($this->featureFlagChecker);
+        $this->assessmentTestSession = $this->createMock(AssessmentTestSession::class);
+    }
+
+    /**
+     * @dataProvider providerProctorRoles
+     */
+    public function testGetPausedStateMessage(array $proctorRoles, string $message): void
+    {
+        $testRunnerMessageService = $this->getClassTestRunnerMessageService($this->serviceManager, ['proctorRoles' => $proctorRoles]);
+        $pausedStateMessage = $testRunnerMessageService->getPausedStateMessageFrom($this->assessmentTestSession);
+
+        self::assertSame($message, $pausedStateMessage);
+    }
+
+    public function testGetPausedStateMessageReturnEmpty(): void
+    {
+        $this->featureFlagChecker->method('isEnabled')->willReturn(true);
+        $testRunnerMessageService = $this->getClassTestRunnerMessageService($this->serviceManager);
+        $pausedStateMessage = $testRunnerMessageService->getPausedStateMessageFrom($this->assessmentTestSession);
+
+        self::assertEmpty($pausedStateMessage);
+    }
+
+    public function providerProctorRoles(): array
+    {
+        return [
+            'message with defined roles' => [
+                (new \common_session_AnonymousSession())->getUserRoles(),
+                'The assessment has been suspended. To resume your assessment, please relaunch it and contact your proctor if required.'
+            ],
+            'message without defined roles' => [
+                [],
+                'The assessment has been suspended. To resume your assessment, please relaunch it.'
+            ]
+        ];
+    }
+
+    /**
+     * @param array $options
+     * @param ServiceManager $serviceManager
+     * @return TestRunnerMessageService
+     */
+    private function getClassTestRunnerMessageService($serviceManager, $options = []): TestRunnerMessageService
```

Comment:
probably we can add type hints to arguments and remove phpdock block, WDYT?

Reply:
Good idea